pkg/containermanager/docker/docker_manager.go

```diff
@@ -80,6 +80,20 @@ func (dc dockerContainerManager) Provision(ctx context.Context, containerConfig
 	return nil
 }
 
+func (dc dockerContainerManager) Deprovision(ctx context.Context, containerConfig *containermanager.ContainerConfig) error {
```

Comment:
why does deprovision take a container config?

Reply:
agreed. looks like all we need is the name (ie. `instanceId`)